docs/architecture.md

```diff
@@ -0,0 +1,46 @@
+# source{d} lookout Architecture
+
+_You will find the project background, goals, constraints, solution details and other info in [**source{d} lookout DesignDoc**](https://docs.google.com/document/d/1pqz-_SHO5BsJE-aa8o_bAY3r5vR67amnWN8-qZc2UgY/edit#)_
+
+![lookout service sequence diagram](assets/lookout-seq-diagram.png)
+
+You can [edit this image](https://mermaidjs.github.io/mermaid-live-editor/#/edit/eyJjb2RlIjoic2VxdWVuY2VEaWFncmFtXG4gICAgcGFydGljaXBhbnQgR2l0SHViXG4gICAgcGFydGljaXBhbnQgU2VydmVyXG4gICAgcGFydGljaXBhbnQgRGF0YVNlcnZpY2VcbiAgICBwYXJ0aWNpcGFudCBCYmxmc2hcbiAgICBwYXJ0aWNpcGFudCBBbmFseXplclxuICAgIGxvb3AgSGVhbHRoY2hlY2tcbiAgICAgICAgU2VydmVyIC0-IFNlcnZlcjoga2VlcCB3YWl0aW5nXG4gICAgZW5kXG4gICAgbG9vcCByZWdpc3RlcmVkIFJlcG9zXG4gICAgICAgIFNlcnZlciAtPj4gR2l0SHViOiBQb2xsOiBQUnMgZXZlbnRzXG4gICAgICAgIGxvb3AgcmVnaXN0ZXJlZCBBbmFseXplcnNcbiAgICAgICAgICAgIFNlcnZlciAtPj4gK0FuYWx5emVyOiAoZ1JQQykgTm90aWZ5UmV2aWV3RXZlbnRcbiAgICAgICAgICAgIEFuYWx5emVyIC0-PiArRGF0YVNlcnZpY2U6IChnUlBDKSBHZXRDaGFuZ2VzIC8gR2V0RmlsZXNcbiAgICAgICAgICAgIG9wdCByZXEuV2FudFVBU1Q_XG4gICAgICAgICAgICAgICAgRGF0YVNlcnZpY2UgLT4-ICtCYmxmc2g6IChnUlBDKSBQYXJzZVxuICAgICAgICAgICAgICAgIEJibGZzaCAtLT4-IC1EYXRhU2VydmljZTogdWFzdC5Ob2RlXG4gICAgICAgICAgICBlbmRcbiAgICAgICAgICAgIERhdGFTZXJ2aWNlIC0tPj4gLUFuYWx5emVyOiBzdHJlYW0gb2YgQ2hhbmdlIC8gRmlsZVxuICAgICAgICAgICAgQW5hbHl6ZXIgLS0-PiAtU2VydmVyOiBDb21tZW50c1xuICAgICAgICBlbmRcbiAgICBTZXJ2ZXIgLT4-IEdpdEh1YjogUG9zdCBhbGwgY29tbWVudHNcbiAgICBlbmRcbiIsIm1lcm1haWQiOnsidGhlbWUiOiJkZWZhdWx0In19) using [mermaid](https://mermaidjs.github.io). ([sourcecode](assets/lookout-seq-diagram.md))
+
+Lookout consists of the following components:
+
+
+## Server
+
+It is the main component, running in a separate process.
+It is responsible for orchestrating all the other services.
+It takes review requests made by an external code review system, calls the registered analyzers to review the changes, and posts the results back.
+
+
+## lookout DataService
```

Comment:
For the same reasoning as above, I would not consider `DataService` a component. I would simply explain this as one of the things that the lookout server does.

This is a personal opinion on what is a _component_. But from an architecture point of view, I would stop at `lookout`.

For someone that deploys lookout, an analyzer needs to call the `lookoutd` component, using the `DataService` gRPC service definition. For the same reason we have the `lookout` component contact GitHub, and not a separate `github provider`, or `go-github` component.

Reply:
I also found that "DataService" is part of the API that the user is using, so it's may worth to explain who's attending their requests and how is that related to Lookout.
In a further iteration, DataService could be even physically separated, and this info would be still meaningful.

Reply:
addressed by 3484af7